src/client/common/application/notebook.ts

```diff
@@ -107,6 +102,9 @@ export class VSCodeNotebook implements IVSCodeNotebook {
     ): Disposable {
         return notebook.registerNotebookKernelProvider(selector, provider);
     }
+    public createNotebookController(id: string, selector: NotebookSelector, label: string, handler?: NotebookExecutionHandler, preloads?: NotebookKernelPreload[]): NotebookController {
```

Comment:
How about making this static, we don't really need an instance method on this (this way we know that there's no state information or the like).
E.g. my immediate assumption was there's something that ties this controller to the instance of this class